### PR TITLE
Java11 Prerequisites

### PR DESCRIPTION
**Pull request by leifker**
The key motivation is to update build process and a couple classes for java 11 compatibility. 

The default JDK must remain java8 until the spark-lineage module upgrades from Spark 2.x as it has a hard dependency on the java 8 jvm. It is not possible to use the `--release` flag in later jvms and run spark 2.4 tests. 

In order to keep java8 as the default and selectively use java11 we upgrade to gradle 6 with support for [toolchains](https://blog.gradle.org/java-toolchains). This allows us to include both java8 and java11 modules, as long as java8 modules do not depend on java11. The reverse is possible, java11 modules can depend on java8 modules.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 buildscript {
   ext.junitJupiterVersion = '5.6.1'
-  ext.pegasusVersion = '28.3.7'
+  ext.pegasusVersion = '29.22.16'
```
Comment (dleifkerz) [inline build.gradle:3]: This was bumped to support gradle 6, the old version did not support it. Decided to conservatively increase this version to what appeared to be the latest stable 29.x without moving to the very latest 30.x. Didn't want to move too fast here.

File: metadata-integration/java/datahub-client/build.gradle [hunk 2]
```diff
@@ -110,6 +110,9 @@ shadowJar {
   relocate 'org.abego.treelayout', 'datahub.shaded.treelayout'
   relocate 'org.slf4j', 'datahub.shaded.slf4j'
   relocate 'javax.annotation', 'datahub.shaded.javax.annotation'
+  relocate 'com.github.benmanes.caffeine', 'datahub.shaded.com.github.benmanes.caffeine'
```
Comment (dleifkerz) [inline metadata-integration/java/datahub-client/build.gradle:113]: These were added due to the bump in pegasus version [here](https://github.com/linkedin/datahub/pull/4340/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7L3).

File: metadata-io/src/main/java/com/linkedin/metadata/search/ranker/SearchRanker.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 /**
  * In memory ranker that re-ranks results returned by the search backend
  */
-public abstract class SearchRanker {
+public abstract class SearchRanker<U extends Comparable<? super U>> {
```
Comment (dleifkerz) [inline metadata-io/src/main/java/com/linkedin/metadata/search/ranker/SearchRanker.java:19]: Compiles with java 11

**Review by RyanHolstien — APPROVED** [review]
LGTM, changes look pretty straightforward. @shirshanka can we get a forced build execution since this is a first time contribution? Assuming the build goes through I think this is good to go. CI shouldn't need any updates for these prereqs.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**